proposals/0000-a11y-cross-platform.md

```diff
@@ -0,0 +1,133 @@
+
+## Detailed design
+
+### accessibilityViewIsModal
+
+#### Android/Fire OS
+
+When a component marks itself as modal, code in the view manager will need to traverse all other components in the view tree and mark them as importantForAccessibility=no. A window content changed event with content change types set to subtree will then need to be fired on the root window to ensure that assistive technologies receive the tree updates.
+
+It may also be necessary to install a view tree observer once the modal component is active to prevent background components from appearing. We'll need to do some real-world testing to understand how realistic this scenario might be, and whether it's worth the performance cost of managing such an observer.
+
+#### iOS
+
+This property is currently supported on iOS, so no change to iOS support is needed.
+
+### accessibilityElementsHidden
+
+#### Android/Fire OS
+
+Supporting this property in Android may be as simple as marking the component as importantForAccessibility=noHideDescendants
+
+If this doesn't match the iOS behavior, we may need to build an accessibility delegate to hide the children. This will likely be tricky, since the children will still be generating events.
+
+#### iOS
+
+This property is already supported on iOS, so no iOS changes are required.
+
+### accessibilityLiveRegion
+
+This allows components to tell assistive technologies how to react to changes in their content.
+
+#### Android/Fire OS
+
+This property is already supported on Android, so no changes in Android support are required.
+
+#### iOS
+
+We believe this functionality can be implemented using accessibility notifications on iOS. For assertive live regions, the updated text of the component can be sent immediately to iOS as an announcement notification. For polite live regions, code can wait for the AnnouncementDidFinish notification before posting the updated text as an announcement notification.
+
+### accessibilityCompoundComponent
+
+We think the current implementation, which is called "accessible", should be renamed to accessibilityCompoundComponent, to more accurately represent what it does. The functionality should remain the same.
+
+### accessibilityRelationships
+
+This property describes how this component is related to others. Relationship types include:
+
+* labelFor: This component labels the specified component
+* labeledBy: this component is labeled by the specified component
+* describedBy: this component is described by the specified components.
+
+When referring to other components, we may need to add an additional property to view-- namely accessibilityId. We need to find the best forward-looking approach for referencing other components.
```

Comment:
We already have `nativeID` that we can use for this

Reply:
Looks like it was introduced for both platforms here in June 2017 https://github.com/facebook/react-native/commit/70e04555226fe3a57eddfb7d601ad9ff600e2b40